### ISSUE TITLE
$.pjax doesn't work, but $.pjax.submit does

### ISSUE THREAD

**Issue by alonewarrior** [defunkt/jquery-pjax#339]
I have been attempting to get PJAX to work on my site for awhile now. It succeeded for awhile last night, but suddenly it no longer wishes to work properly. Even when I adjust the timeout, 2 requests still occur, with the latter not containing the appropriate X-PJAX information and causing a full page reload.

This is what I'm using to call PJAX:

``` js
$(function(){
    $(document).pjax('a.pjax', '#test', { fragment: ('#test'), timeout: 10000 });
});
```

On the other hand, the form portion works perfectly fine, and I haven't had any issues yet today.

I am at a loss of what to do right now; I'm going insane!

**Comment by kirkbushell** [comment]
^ I'm finding this as well. It'll work one day then the next it's broken, it seems fickle.

**Comment by mislav** [comment]
Sorry that all of you have troubles with pjax. The behavior of pjax varies pretty wildly with what sort of response it gets from the server. For example, the original poster defined the `fragment: "#test"` option which means that the element with ID "test" will be extracted as pjax content from the server response. If that element doesn't exist, however, the page will be force-reloaded.

So when you're reporting a pjax bug, please include the full code of how you set up pjax on your site (like the original poster), what was the behavior that you expected, what is the exact behavior that you got, and the raw HTML response from the server for the pjax request. Thanks!
